### PR TITLE
Initial CheckIn of Validator tool and usage in the Upgrade pipeline

### PR DESCRIPTION
**Pull request by ddrechse**
# Description

Please include a summary of the change and which issue is fixed.  If there are any dependencies, for example on a PR in another repository, please list those.

 VZ-3445 Upgrade tests should check if new image matches BOM
 Initial checkin of Bom-Validator tool
 Builds in verrazzano pipeline
 Validates in upgrade pipeline
 Modeled after analysis Tool pipeline build and usage semantics

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -203,6 +203,26 @@ pipeline {
...
+            post {
+                failure {
+                    script {
+                        SKIP_TRIGGERED_TESTS = true
+                    }
+                }
+                success {
+                    sh """
+                        cd ${GO_REPO_PATH}/verrazzano/tools/bom-validator
+                        oci --region us-phoenix-1 os object put --force --namespace ${OCI_OS_NAMESPACE} -bn ${OCI_OS_BUCKET} --name ${env.BRANCH_NAME}/bom-validator --file ./out/linux_amd64/bom-validator
```
Comment (michael-cico) [inline Jenkinsfile:219]: Not necessarily in this pass, but do we need to publish a Mac version of this like the analysis tool?  Probably only if we intend to publish it as a release artifact.

Comment (michael-cico) [reply Jenkinsfile:219]: nm, looks like I spoke too soon, I see you build one already.

File: tools/bom-validator/bom-validator.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// Struct based on Verrazzano BOM JSON
+type verrazzanoBom struct {
```
Comment (michael-cico) [inline tools/bom-validator/bom-validator.go:17]: Also not for this pass, but if you want to use it I moved the VPO BOM code into `pkg/bom` at the top-level in the repo.

**Review by michael-cico — APPROVED** [review]

File: tools/bom-validator/bom-validator.go (hunk 2 above)
Comment (ddrechse) [reply tools/bom-validator/bom-validator.go:17]: Agreed, this task captures that
VZ-3553 BomValidator: write unit tests

File: Jenkinsfile (hunk 1 above)
Comment (ddrechse) [reply Jenkinsfile:219]: Maybe, the makefile builds one so if its needed it is being built

**Pull request merged**